soccer/NewRefereeModule.cpp

```diff
@@ -186,17 +189,35 @@ void NewRefereeModule::run() {
         ballPlacementx = packet->wrapper.designated_position().x();
         ballPlacementy = packet->wrapper.designated_position().y();
 
-        std::string yellow_name = yellow_info.name;
-        for (char& letter : yellow_name) {
-            letter = tolower(letter);
+        // If we have no name, we're using a default config and there's no
+        // sense trying to match the referee's output (because chances are
+        // everything is just on default configuration with no names, so more
+        // than one team/soccer instance will be trying to use the same color)
+        if (Team_Name_Lower.length() > 0) {
+            // We only want to change teams if we get something that actually
+            // matches our team name (either yellow or blue).
+            // Otherwise, keep the current color.
+            if (boost::iequals(yellow_info.name, Team_Name_Lower)) {
```

Comment:
I think this could just be `yellow_info.name == Team_Name_Lower`

Reply:
iequals is case-insensitive comparison. I don't think we have a guarantee that `yellow_info.name` from the packet will always be lower case, do we?

Reply:
Oh rip, should have read the second sentence of the two sentence doc page that talked about it being case-insensitive. 100% better to assume not and just deal with it